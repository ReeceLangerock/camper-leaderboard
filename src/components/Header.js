import React from 'react'

export class Header extends React.Component {
  render () {
    return (
      <div className='Header' ><h1>FCC Camper Leaderboard</h1></div>
    )
  }
}
export default Header
