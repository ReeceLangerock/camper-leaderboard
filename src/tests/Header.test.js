import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, find, render, simulate } from "enzyme";
import enzymeSerializer from "enzyme-to-json/serializer";

import Header from "./../components/Header";

expect.addSnapshotSerializer(enzymeSerializer);

describe("Header", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Header />, div);
  });

  let component;
  beforeEach(() => {
    const props = {};
    component = mount(<Header {...props} />);
  });

  it("renders page title", () => {
    const title = <h1>FCC Camper Leaderboard</h1>;
    expect(component.contains(title)).toEqual(true);
  });

  it("has the correct class", () => {
    expect(component.hasClass("Header")).toEqual(true);
  });
});

describe("Snapshot Header", () => {
  it("matches snapshot", () => {
    const comp = mount(<Header />);
    expect(comp).toMatchSnapshot();
  });
});
