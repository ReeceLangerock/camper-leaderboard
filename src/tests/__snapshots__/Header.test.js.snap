// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Header matches snapshot 1`] = `
<Header>
  <div
    className="Header"
  >
    <h1>
      FCC Camper Leaderboard
    </h1>
  </div>
</Header>
`;
